ui/core-react/src/test/loading/LoadingPrompt.test.tsx

```diff
@@ -2,50 +2,52 @@
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
-import { mount, shallow } from "enzyme";
+import { render, screen } from "@testing-library/react";
+import { expect } from "chai";
 import * as React from "react";
 import { LoadingPrompt } from "../../core-react";
 
 describe("<LoadingPrompt />", () => {
-  it("should render", () => {
-    const wrapper = mount(
-      <LoadingPrompt />,
-    );
-    wrapper.unmount();
-  });
-
-  it("renders correctly", () => {
-    shallow(
-      <LoadingPrompt />,
-    ).should.matchSnapshot();
-  });
-
   it("renders with text correctly", () => {
-    shallow(<LoadingPrompt title="title" />).should.matchSnapshot();
+    render(<LoadingPrompt title="title" />);
+
+    expect(screen.getByText("title")).to.exist;
   });
 
   it("renders with text and message correctly", () => {
-    shallow(<LoadingPrompt title="title" message="description" />).should.matchSnapshot();
+    render(<LoadingPrompt title="title" message="description" />);
+
+    expect(screen.getByText("description")).to.exist;
   });
 
   it("renders with indeterminate ProgressBar", () => {
-    shallow(<LoadingPrompt showIndeterminateBar />).should.matchSnapshot();
+    const {container} = render(<LoadingPrompt showIndeterminateBar />);
+
+    expect(container.querySelector(".iui-indeterminate")).to.exist;
   });
 
   it("renders with text and message, and determinate", () => {
-    shallow(<LoadingPrompt title="title" message="description" isDeterminate={true} />).should.matchSnapshot();
+    const {container} = render(<LoadingPrompt title="title" message="description" isDeterminate={true} />);
+
+    expect(container.querySelector<HTMLDivElement>(".lb-container > .fill")?.style).to.include({width: "0%"});
   });
 
   it("renders with text and message, and determinate and percent", () => {
-    shallow(<LoadingPrompt title="title" message="description" isDeterminate={true} percent={50} />).should.matchSnapshot();
+    const {container} = render(<LoadingPrompt title="title" message="description" isDeterminate={true} percent={50} />);
+
+    expect(container.querySelector<HTMLDivElement>(".lb-container > .fill")?.style).to.include({width: "50%"});
   });
 
   it("renders with text and message, and determinate and showCancel", () => {
-    shallow(<LoadingPrompt title="title" message="description" isDeterminate={true} percent={50} showCancel={true} />).should.matchSnapshot();
+    render(<LoadingPrompt title="title" message="description" isDeterminate={true} percent={50} showCancel={true} />);
+
+    expect(screen.getByRole("button", {name: "Cancel"})).to.exist;
```

Comment:
This "Cancel" is hardcoded and not localized, we might want to localize it if this component should survive, the test will need to be updated with the name being the localized string, instead of the English text.

Reply:
I did not think we did any localized strings in the tests, but I could be forgetting it.

Reply:
There is really few elements that are localized in core react, one of them was the ThemedSelect, and when in test, the localization system simply returns the key, which can then be tested (there was indeed a typo in the localization key, "reactelect.noOptions". This one is simply not using localization, so I cant check for "loadingPrompt.cancel".

Reply:
Ugh. Good riddance to react-select. It's gone in 4.0.